packages/std/src/os/kernel.rs

```diff
@@ -171,10 +171,7 @@ pub enum IbcExecuteMsg {
         amp_packet: AMPPkt,
     },
     SendMessageWithFunds {
-        recipient: AndrAddr,
-        message: Binary,
-        funds: Coin,
-        original_sender: String,
+        amp_packet: AMPPkt,
```

Comment:
I'm unsure about passing the entire packet here. I think sending the specific data we need may be safer? Wdyt? This would mean adding previous hops and username.